### PR TITLE
fix three bugs about dfs and ethernetif

### PR DESCRIPTION
**Pull request by longtengmcu**
## 拉取/合并请求描述：(PR description)

[
1、解决do_pollfd函数中处理底层网络设备返回错误-1的情况，返回-1时select函数也要返回-1 
2、网卡设备调用关闭dhcp时，底层要调用dhcp_stop函数来关闭dhcp功能，打开时再调用dhcp_start，否则再次打开dhcp时无法获取ip地址或者出现在使用静态ip地址会在dhcp租约到期时自动再获取ip地址。
3、sal_socket.c程序中的sal_accept函数增加判断网卡up,down的功能，当网卡down时，sal_accept函数能立即返回错误。
]

以下的内容不应该在提交PR时的message修改，修改下述message，PR会被直接关闭。请在提交PR后，浏览器查看PR并对以下检查项逐项check，没问题后逐条在页面上打钩。
The following content must not be changed in submitted PR message. Otherwise, the PR will be closed immediately. After submitted PR, please use web browser to visit PR, and check items one by one, and ticked them if no problem.

### 当前拉取/合并请求的状态 Intent for your PR

必须选择一项 Choose one (Mandatory):

- [ ] 本拉取/合并请求是一个草稿版本 This PR is for a code-review and is intended to get feedback
- [x] 本拉取/合并请求是一个成熟版本 This PR is mature, and ready to be integrated into the repo

### 代码质量 Code Quality：

我在这个拉取/合并请求中已经考虑了 As part of this pull request, I've considered the following:

- [x ] 已经仔细查看过代码改动的对比 Already check the difference between PR and old code
- [x] 代码风格正确，包括缩进空格，命名及其他风格 Style guide is adhered to, including spacing, naming and other style
- [x] 没有垃圾代码，代码尽量精简，不包含`#if 0`代码，不包含已经被注释了的代码 All redundant code is removed and cleaned up
- [x] 所有变更均有原因及合理的，并且不会影响到其他软件组件代码或BSP All modifications are justified and not affect other components or BSP
- [x] 对难懂代码均提供对应的注释 I've commented appropriately where code is tricky
- [x] 本拉取/合并请求代码是高质量的 Code in this PR is of high quality

### CODE REVIEW

File: components/dfs/src/poll.c [hunk 1]
```diff
@@ -137,6 +137,14 @@ static int do_pollfd(struct pollfd *pollfd, rt_pollreq_t *req)
                 req->_key = pollfd->events | POLLERR | POLLHUP;
 
                 mask = f->fops->poll(f, req);
+
+                /* dealwith the device return error -1 zhaoshimin 20200509*/
```
Comment (BernardXiong) [inline components/dfs/src/poll.c:141]: Please not put your name here.

File: components/net/lwip-2.0.2/src/netif/ethernetif.c [hunk 2]
```diff
@@ -164,6 +164,16 @@ static int lwip_netdev_set_dns_server(struct netdev *netif, uint8_t dns_num, ip_
 static int lwip_netdev_set_dhcp(struct netdev *netif, rt_bool_t is_enabled)
 {
     netdev_low_level_set_dhcp_status(netif, is_enabled);
+
+	/*zhaoshimin 20200508 add dhcp start or stop must call dhcp_start and dhcp_stop function*/
```
Comment (BernardXiong) [inline components/net/lwip-2.0.2/src/netif/ethernetif.c:168]: Please not put your name here.

**Comment by BernardXiong** [comment]
Please not put your name in the code, only put in changelog.

**Comment by longtengmcu** [comment]
OK


fhqlongteng@163.com
 
From: Bernard Xiong
Date: 2020-05-17 17:03
To: RT-Thread/rt-thread
CC: longteng; Author
Subject: Re: [RT-Thread/rt-thread] fix three bugs about dfs and ethernetif (#3606)
@BernardXiong commented on this pull request.


In components/net/lwip-2.0.2/src/netif/ethernetif.c:
> @@ -164,6 +164,16 @@ static int lwip_netdev_set_dns_server(struct netdev *netif, uint8_t dns_num, ip_
 static int lwip_netdev_set_dhcp(struct netdev *netif, rt_bool_t is_enabled)
 {
     netdev_low_level_set_dhcp_status(netif, is_enabled);
+
+	/*zhaoshimin 20200508 add dhcp start or stop must call dhcp_start and dhcp_stop function*/

Please not put your name here.

File: components/net/lwip-2.0.2/src/netif/ethernetif.c (hunk 2 above)
Comment (longtengmcu) [reply components/net/lwip-2.0.2/src/netif/ethernetif.c:168]: OK

File: components/dfs/src/poll.c (hunk 1 above)
Comment (longtengmcu) [reply components/dfs/src/poll.c:141]: OK

**Pull request closed without merging**